src/test/java/org/ays/auth/service/impl/AysRoleUpdateServiceImplTest.java

```diff
@@ -740,6 +760,44 @@ void givenValidId_whenRoleUsing_thenThrowAysRoleAssignedToUserException() {
         Mockito.verify(roleReadPort, Mockito.times(1))
                 .isRoleUsing(Mockito.anyString());
 
+        Mockito.verify(identity, Mockito.times(1))
+                .getInstitutionId();
+
+        Mockito.verify(roleSavePort, Mockito.never())
+                .save(Mockito.any(AysRole.class));
+    }
+
+    @Test
+    void givenValidId_whenRoleNotMatchedWithInstitution_thenThrowAysRoleNotExistByIdException() {
+
+        // Given
+        String mockId = "32657ecb-5172-401d-8b29-a8e4454e8243";
+        String mockRandomInstitutionId = "3b9dcab2-028f-4bb8-9c9b-23130d9579ae";
+
+        // When
+        AysRole mockRole = new AysRoleBuilder()
+                .withValidValues()
+                .withId(mockId)
+                .build();
+        Mockito.when(roleReadPort.findById(Mockito.anyString()))
+                .thenReturn(Optional.of(mockRole));
+
+        Mockito.when(identity.getInstitutionId())
+                .thenReturn(mockRandomInstitutionId);
```

Comment:
Bu bilgi dışarıdan gelmediği için `Given` alanına yazmamalıyız
```suggestion

        // When
        AysRole mockRole = new AysRoleBuilder()
                .withValidValues()
                .withId(mockId)
                .build();
        Mockito.when(roleReadPort.findById(Mockito.anyString()))
                .thenReturn(Optional.of(mockRole));

        Mockito.when(identity.getInstitutionId())
                .thenReturn("3b9dcab2-028f-4bb8-9c9b-23130d9579ae");
```